### PR TITLE
(chore) Fix TypeScript errors when locally linking the form engine

### PR DESCRIPTION
**Pull request by samuelmale**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

- Aligns i18n related libraries to what is supported in the framework
- Tweaks to the TS config

## Related Issue

Error:
Screenshot 2024-12-04 at 17 42 39 (1)

### CODE REVIEW

**Review by ibacher — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -93,8 +93,8 @@
     "eslint-plugin-react-hooks": "^4.6.2",
     "eslint-plugin-testing-library": "^6.2.2",
     "husky": "^8.0.3",
-    "i18next": "^23.16.4",
-    "i18next-parser": "^9.0.2",
+    "i18next": "21.10.0",
+    "i18next-parser": "8.13.0",
```
Comment (NethmiRodrigo) [inline package.json:97]: i18n versions are a bit finicky, these versions currently give this error if you run the extract-translations command. 
Screenshot 2024-12-05 at 9 55 48 PM

Comment (NethmiRodrigo) [reply package.json]: Pushed a commit that updates the parser version

**Review by denniskigen — APPROVED** [review]

**Comment by NethmiRodrigo** [comment]
Unforunately, bumping up the parser gives the type issues @ibacher 
Screenshot 2024-12-06 at 12 35 01 AM

**Comment by samuelmale** [comment]
@NethmiRodrigo I would revert the version to `v8.13.0`

File: package.json (hunk 1 above)
Comment (NethmiRodrigo) [reply package.json]: @samuelmale reverting it would give this issue

Comment (samuelmale) [reply package.json]: Ohh yes, I had missed that! How about updating the parser version across the framework?

**Comment by NethmiRodrigo** [comment]
> Unforunately, bumping up the parser gives the type issues @ibacher @denniskigen @samuelmale Screenshot 2024-12-06 at 12 35 01 AM

As it clearly shows in the screenshot, I am an idiot and checked out out to main before testing it. Its all good to go, this error doesn't come up anymore with the updated parser and the i18n version that Sam updated it to. Merging this in. Thanks @samuelmale

**Pull request merged**